test/util/mock-server.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ** This file is automatically generated by gapic-generator-typescript. **
+// ** https://github.com/googleapis/gapic-generator-typescript **
+// ** All changes to this file may be overwritten. **
+
+import {describe, it} from 'mocha';
+import {MockServer} from '../../src/util/mock-servers/mock-server';
+import * as net from 'net';
+import * as assert from 'assert';
+
+interface ServerError {
+  code: string;
+}
+
+// TODO: Test server shuts down
+
+describe('Bigtable/Mock-Server', () => {
+  const inputPort = '1234';
+  let server: MockServer;
+  function checkPort(port: string, inUse: boolean, callback: () => void) {
+    const netServer = net.createServer();
+    netServer.once('error', (err: ServerError) => {
+      if (inUse) {
+        assert.strictEqual(err.code, 'EADDRINUSE');
+      } else {
+        assert.notEqual(err.code, 'EADDRINUSE');
+      }
+      netServer.close();
+      callback();
+    });
+    netServer.once('listening', () => {
+      // close the server if listening doesn't fail
+      netServer.close();
+    });
+    // netServer.listen(port);
+    netServer.listen(`localhost:${port}`);
+  }
+  describe('Ensure server shuts down properly when destroyed', () => {
```

Comment:
If I add a before hook with the following in it:

`checkPort(inputPort, false, done);`

the `assert.notEqual(err.code, 'EADDRINUSE');` fails, but I am not sure why at this time.